### PR TITLE
A Multisig of Signers: Small tidbits to complete new signing flow

### LINKED ISSUE #2408: In-wallet swaps - approval limit can't be changed
### Discord Discussion Link

https://discord.com/channels/808358975287722045/888500174685614090/1029325326385422336

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

Change the spend limit

### What did not work?

It always jumps back to infinite.

image

### Version

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #2482: Unlock signing screen visual issue
image

To recreate
- Ensure wallet is locked
- Set up an in-wallet swap on Ethereum
- Attempt to execute swap
- See the above screen

### PR DESCRIPTION
**Pull request by Shadowfiend**
This completes a few small tidbits for the new signing flow to be ready to be switched on:
- Some missing i18n.
- Some small styling fixes.
- Asset swaps summary has basic swap information.
- Asset swaps do contract name enrichment.

In the process, re-enriching already-enriched transactions is now working again (fixes #2408, which was broken in #2331). Two more small fixes:
- `SharedIcon` only generates `button` if click handlers are provided.
- `SingleAsset` page correctly i18ns the link to a scanner.

The goal is to flip on the `USE_UPDATED_SIGNING` flag if this PR is good to go. Any other tidbits that are required should be catalogued here.

## Testing Environment

```
USE_UPDATED_SIGNING_UI=true
```

Closes #2482.

Latest build: [extension-builds-2461](https://github.com/tallycash/extension/suites/9073522155/artifacts/419982277) (as of Tue, 01 Nov 2022 18:17:38 GMT).

### CODE REVIEW

File: background/services/enrichment/transactions.ts [hunk 1]
```diff
@@ -156,22 +157,29 @@ export default async function resolveTransactionAnnotation(
...
-    blockTimestamp: undefined,
-    timestamp: Date.now(),
-    type: "contract-deployment",
-    transactionLogoURL: assets.find(
-      (asset) =>
-        asset.metadata?.logoURL &&
-        asset.symbol === transaction.network.baseAsset.symbol
-    )?.metadata?.logoURL,
-  }
+  // By default, annotate all requests as contract interactions, unless they
```
Comment (0xDaedalus) [inline background/services/enrichment/transactions.ts:168]: I have a strong desire to rework the construction logic in `resolveTransactionAnnotation` from a gradual buildup of a `TransactionAnnotation` object through in-order mutation of the object to something more like a reducer function with all logic for a given type of annotation grouped in its own function.  Out of scope of this PR of course but just wanted to write down my thoughts somewhere.

File: ui/components/Signing/SignatureDetails/TransactionSignatureDetails/RawDataPanel.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ export default function RawDataPanel({
 
   const copyData = () => {
     navigator.clipboard.writeText(input ?? "")
-    dispatch(setSnackbarMessage("Raw data copied to clipboard"))
+    dispatch(setSnackbarMessage(t("rawDataCopyMsg")))
```
Comment (0xDaedalus) [inline ui/components/Signing/SignatureDetails/TransactionSignatureDetails/RawDataPanel.tsx:20]: ❤️

File: ui/components/Signing/Signer/SignerLedger/SignerLedgerFrame.tsx [hunk 3]
```diff
@@ -15,10 +15,11 @@ export default function SignerLedgerFrame<T extends SignOperationType>({
   request,
   signer,
   signingAddress,
-  signingActionLabel,
+  signingActionLabelI18nKey,
   signActionCreator,
   rejectActionCreator,
 }: SignerFrameProps<T>): ReactElement {
+  const { t: globalT } = useTranslation()
```
Comment (greg-nagy) [inline ui/components/Signing/Signer/SignerLedger/SignerLedgerFrame.tsx:22]: must resist the urge to suggest something silly ... like let's rename this to MrT

CleanShot 2022-10-20 at 16 43 05

**Review by greg-nagy — COMMENTED** [review]
my most meaningful review ... ever

File: background/services/enrichment/transactions.ts (hunk 1 above)
Comment (Shadowfiend) [reply background/services/enrichment/transactions.ts:168]: Yeah 100% agree, this is extremely messy and not particularly scalable.

File: ui/components/Signing/SignatureDetails/index.tsx [hunk 4]
```diff
@@ -32,7 +32,15 @@ export type ResolvedSignatureDetails = {
   signer: AccountSigner
   signingAddress: AddressOnNetwork
   renderedSigningData: ReactElement
-  signingActionLabel: string
+  /**
+   * A string that represents what signing this data will achieve. Some
+   * signers may ignore this string, others may use it for their confirmation
+   * button.
+   */
+  signingActionLabelI18nKey:
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply ui/components/Signing/SignatureDetails/index.tsx:40]: Correct. The other option was pre-resolving the i18n outside a component, but that was… Weirder.

**Comment by 0xDaedalus** [comment]
@Shadowfiend pending merge conflict resolution is this ready to be un-drafted?

**Comment by Shadowfiend** [comment]
I think I still need to fix the issue you flagged in #2482 and resolve merge conflicts, then we can merge and move to feature flag flip  + QA testing.

**Comment by Shadowfiend** [comment]
Ok, this fellow is ready for review.

**Pull request merged**